transaction/core/src/tx.rs

```diff
@@ -481,6 +481,17 @@ impl TxOut {
             .as_mut()
             .ok_or(TxOutConversionError::UnknownMaskedAmountVersion)
     }
+
+    /// Check if a TxOut matches another TxOut, except possibly in the
+    /// masked_amount. This is used in MCIP #42 partial fills rules
+    /// verification.
+    pub fn matches_ignoring_amount(&self, other: &TxOut) -> bool {
```

Comment:
Nit: rename to `eq_ignoring_amount`, I think `matches` is ambiguous in this context.